internal/scrapertest/mask.go

```diff
@@ -61,3 +47,78 @@ func maskDataPointSliceValues(dataPoints pdata.NumberDataPointSlice) {
 		dataPoint.SetDoubleVal(0)
 	}
 }
+
+// IgnoreAttributeValue is a CompareOption that clears all values
+func IgnoreAttributeValue(attributeName string, metricNames ...string) CompareOption {
+	return ignoreAttributeValue{
+		attributeName: attributeName,
+		metricNames:   metricNames,
+	}
+}
+
+type ignoreAttributeValue struct {
+	attributeName string
+	metricNames   []string
+}
+
+func (opt ignoreAttributeValue) apply(expected, actual pdata.MetricSlice) {
+	maskMetricSliceAttributeValues(expected, opt.attributeName, opt.metricNames...)
+	maskMetricSliceAttributeValues(actual, opt.attributeName, opt.metricNames...)
+}
+
+// maskMetricSliceAttributeValues sets the value of the specified attribute to
+// the zero value associated with the attribute data type.
+// If metric names are specified, only the data points within those metrics will be masked.
+// Otherwise, all data points with the attribute will be masked.
+func maskMetricSliceAttributeValues(metrics pdata.MetricSlice, attributeName string, metricNames ...string) {
+	if len(metricNames) == 0 {
+		for i := 0; i < metrics.Len(); i++ {
+			dps := getDataPointSlice(metrics.At(i))
+			maskDataPointSliceAttributeValues(dps, attributeName)
+
+			// If attribute values are ignored, some data points may become
+			// indistinguishable from each other, but sorting by value allows
+			// for a reasonably thorough comparison and a deterministic outcome.
+			dps.Sort(func(a, b pdata.NumberDataPoint) bool {
+				if a.IntVal() < b.IntVal() {
+					return true
+				}
+				if a.DoubleVal() < b.DoubleVal() {
+					return true
+				}
+				return false
+			})
+		}
+		return
+	}
+
+	metricNameSet := make(map[string]bool, len(metricNames))
+	for _, metricName := range metricNames {
+		metricNameSet[metricName] = true
+	}
+	for i := 0; i < metrics.Len(); i++ {
+		if metric := metrics.At(i); metricNameSet[metric.Name()] {
+			dps := getDataPointSlice(metrics.At(i))
+			maskDataPointSliceAttributeValues(dps, attributeName)
+		}
+	}
+
+}
+
+// maskDataPointSliceAttributeValues sets the value of the specified attribute to
+// the zero value associated with the attribute data type.
+func maskDataPointSliceAttributeValues(dataPoints pdata.NumberDataPointSlice, attributeName string) {
+	for i := 0; i < dataPoints.Len(); i++ {
+		attributes := dataPoints.At(i).Attributes()
+		attribute, ok := attributes.Get(attributeName)
+		if ok {
+			switch attribute.Type() {
+			case pdata.AttributeValueTypeString: // ok
+			default:
+				panic(fmt.Sprintf("data type not supported: %s", attribute.Type()))
+			}
+		}
+
+		attributes.UpsertString(attributeName, "")
```

Comment:
Do we need `Upsert`? Isn't it expected that the attribute is always set? Metrics builder currently doesn't support recording datapoints without providing all attributes specified in metadata.yaml.

Reply:
I was not aware, but it is a good point even if used without metadata.yaml. We should only ignore if the value is present. If entirely absent, this is notable difference in expectation and comparison should fail.